recipes/bitserializer/all/conanfile.py

```diff
@@ -81,12 +101,27 @@ def source(self):
         get(self, **self.conan_data["sources"][self.version],
             destination=self.source_folder, strip_root=True)
 
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_CPPRESTJSON_ARCHIVE"] = self.options.with_cpprestsdk
```

Comment:
How does those options affect the package? Are they needed in headers? Is there a define passed to the compiler? In case yes, you need to pass them by cpp_info.defines too.

Reply:
> How does those options affect the package?
They are used for select which components will be built and installed (affects also the headers list).

> Are they needed in headers?
No, they are used only in CMake.

> Is there a define passed to the compiler? In case yes, you need to pass them by cpp_info.defines too.
These defines are not necessary in the output package and as I understand they are not reqired in the 'cpp_info.defines'.

Reply:
> They are used for select which components will be built and installed (affects also the headers list).

Do you mean you install different headers according to the options? If yes, then the package should be affected by options and can not use `self.info.clear()`

Reply:
Got it, fixed.